ipywidgets/static/widgets/js/widget.js

```diff
@@ -471,12 +471,20 @@ define(["nbextensions/widgets/widgets/js/manager",
             }, this);
 
             this.options = parameters.options;
+            var that = this;
+            /*
+             * this._displayed is a promise that resolves when the view is
+             * inserted in the DOM.
+             */
+            this.displayed = new Promise(function(resolve, reject) {
+                that.once('displayed', resolve);
+            });
             this.on('displayed', function() { 
```

Comment:
Can we use the Promise here? `this.displayed.then(function() {that._is_displayed = true;})`